app/components/review-warning/component.js

```diff
@@ -0,0 +1,34 @@
+import Component from '@ember/component'
+import { inject as service } from '@ember/service'
+import moment from 'moment'
+
+export default Component.extend({
+  isVisible: false,
+  session: service('session'),
+  store: service('store'),
+  startOfMonth: moment()
+    .startOf('month')
+    .subtract(1, 'months'),
+  endOfMonth: moment()
+    .endOf('month')
+    .subtract(1, 'months'),
```

Comment:
This should actually be called on `didInsertElement` rather than at the time the component definition is called

Reply:
Updated that but I have used `init` instead of `didInsertElement`. In my opinion using `didInsertElement` only makes sense if reports are queried in `didInsertElement` as well because `startOfMonth` and `endOfMonth` need to initialized before querying reports.
What is your opinion on that?

Reply:
@anehx Just discussed with @aziiee, and the `init` hook seems fine to both of us. Feel free to correct us though :wink: